### ISSUE TITLE
The wrapper script should honor http(s)_proxy variables

### ISSUE THREAD

**Issue by sschuberth** [gradle/gradle#11065]
### Expected Behavior

Running `gradlew` on a (Linux / Mac) system that has the `http_proxy` and / or `https_proxy` environment variables set should be able to download the configured Gradle distribution though the specified proxy.

### Current Behavior

The `gradlew` script does not pass any proxy configuration from the `http_proxy` / `https_proxy` environment variables to the JVM that runs the Java implementation for the wrapper, resulting in the download of the Gradle distribution to fail on systems that can only access the Internet via a proxy.

### Context

In environments behind a corporate proxy running a Gradle project's `./gradlew` fails out of the box because the JVM does not know about the proxy even if `http_proxy` / `https_proxy` is set. A Gradle user has to manually pass the `http(s).proxy*` JVM system properties either directly to `gradlew` on the command line via `-D`, or configure them in the `~/.gradle/gradle.properties` file.

This is inconvenient extra work and not in line with other package managers / build systems like e.g. Yarn which do respect `http_proxy` / `https_proxy`.

As such it would be nice if the `gradlew` script would by default parse `http_proxy` / `https_proxy` and pass the information on via the `http(s).proxy*` system properties to the JVM that runs the Java implementation for the wrapper.

**Comment by augi** [comment]
IMHO still valid.

**Comment by gnaggnoyil** [comment]
@sschuberth I saw you once committed code that honor http(s)_proxy env vars in PR #12598, but later force pushed and removed those code. Do you still think it's a good idea to honor environment variables?

**Comment by sschuberth** [comment]
Yes, I still think honoring the `http(s)_proxy` environment variables is a good idea. I just didn't want to do too many things at a time and [keep the amount of changes to a minimum](https://github.com/gradle/gradle/pull/12598#issuecomment-633215270) in that PR. And after my PR I didn't really need that feature anymore as I could use Gradle properties instead; honoring the environment variables would have just been convenience.

**Comment by HackToday** [comment]
if environment proxy variables be honored would be better, right now, seems need pass to program like  `-Dhttp.proxyHost`  etc.

**Comment by jbartok** [comment]
Thank you for your interest in Gradle!

This feature request is in the backlog of the relevant team, but this area of Gradle is currently not in focus. It might take a while before it gets implemented.

**Comment by chkpnt** [comment]
I've just stumbled over this issue during the migration from Docker Desktop to Podman Desktop on macOS. While docker is using a transparent proxy for its containers, Podman is setting the common http_proxy environment variables. So the official Gradle containers can be used out-of-the-box without any further proxy configuration, but not with Podman.

**Comment by Nek-12** [comment]
Stumbled over this when trying to make Gradle work on XCode cloud for a Kotlin Multiplatform project. Would love to get this addressed.

**Comment by cunhaax** [comment]
We're almost in 2026 and this is still a problem.
Now I'm hitting this when running Claude's sandbox environment...

**Comment by nroach44** [comment]
I just lost hours of personal time trying to figure out why a cordova project wouldn't build inside of a CI runner because of this.
